src/bundle/Resources/public/js/scripts/fieldType/ezinteger.js

```diff
@@ -0,0 +1,52 @@
+(function (global) {
+    const SELECTOR_FIELD = '.ez-field-edit-ezinteger';
+
+    class EzInteger extends global.eZ.baseFieldValidator {
+        validateInteger(event) {
+            const isRequired = event.target.required;
+            const value = +event.target.value;
+            const isEmpty =  !event.target.value && event.target.value !== '0';
+            const isInteger = Number.isInteger(value);
+            const isLess = value < parseInt(event.target.getAttribute('min'), 10);
+            const isGreater = value > parseInt(event.target.getAttribute('max'), 10);
+            const isError = isEmpty && isRequired || !isInteger || isLess || isGreater;
+            const label = event.target.closest(SELECTOR_FIELD).querySelector('.form-control-label').innerHTML;
+            let errorMessage;
+
+            if (isEmpty) {
+                errorMessage = global.eZ.errors.emptyField.replace('{fieldName}', label);
+            } else if (!isInteger) {
+                errorMessage = global.eZ.errors.isNotInteger.replace('{fieldName}', label);
+            } else if (isLess) {
+                errorMessage = global.eZ.errors.isLess
+                    .replace('{fieldName}', label)
+                    .replace('{minValue}', event.target.getAttribute('min'));
+            } else if (isGreater) {
+                errorMessage = global.eZ.errors.isGreater
+                    .replace('{fieldName}', label)
+                    .replace('{maxValue}', event.target.getAttribute('max'));
+            }
+
+            return {
+                isError: isError,
+                errorMessage: errorMessage,
+            };
+        }
+    }
+
+    const validator = new EzInteger({
```

Comment:
I prefer to call it: `EzIntegerValidator`. `EzInteger` could imply that it renders the `EzInteger` field instance. The same comment will apply to the similar code above.